### PR TITLE
LicenseClassification: Ensure `licensesByCategory` contains all keys

### PR DESCRIPTION
**Pull request by fviernau**
Make the `Map` contain all category names as keys in order to simplify
its use.

### CODE REVIEW

File: model/src/main/kotlin/licenses/LicenseClassifications.kt [hunk 1]
```diff
@@ -56,6 +56,10 @@ data class LicenseClassifications(
                     getOrPut(category) { mutableSetOf() } += license.id
                 }
             }
+
+            categories.forEach { category ->
```
Comment (sschuberth) [inline model/src/main/kotlin/licenses/LicenseClassifications.kt:60]: > Make the `Map` contain all category names as keys in order to simplify its use.

I guess the more important thing here is to be able to distinguish map look-ups with a non-existing category as the key vs. an existing category that has no licenses assigned.

Here's a more compact implementation suggestion:

    val licensesByCategory: Map<String, Set<SpdxSingleLicenseExpression>> by lazy {
        buildMap<String, MutableSet<SpdxSingleLicenseExpression>> {
            categorizations.forEach { license ->
                categoryNames.forEach { category ->
                    val licenses = getOrPut(category) { mutableSetOf() }
                    if (category in license.categories) licenses += license.id
                }
            }
        }
    }

Comment (sschuberth) [reply model/src/main/kotlin/licenses/LicenseClassifications.kt:60]: Also, please add a test-case in `LicenseClassificationsTest`'s `"licensesByCategory" should` test case that highlights that for existing but empty categorizations now an empty set instead of `null` is returned (above `"return null when querying the licenses for an unknown category")`.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/licenses/LicenseClassifications.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/licenses/LicenseClassifications.kt:60]: > I guess the more important thing here is to be able to distinguish map look-ups with a non-existing category as the key vs. an existing category that has no licenses assigned.

Is this a general remark, or are you saying the commit message is not detailed enough and must be changed?

Comment (sschuberth) [reply model/src/main/kotlin/licenses/LicenseClassifications.kt:60]: It's a general remark. But personally, I would have mentioned (only) that in the commit message. IMO it's clearer than saying "to simplify its use" without an example of how the use would have looked before and after the change.

TL;DR no strict need to change the commit message, just a suggestion.

File: model/src/test/kotlin/licenses/LicenseClassificationsTest.kt [hunk 2]
```diff
@@ -63,7 +64,7 @@ class LicenseClassificationsTest : WordSpec({
             exception.message shouldContain "[ASL-1]"
         }
 
-        "detect licenses referencing non-existing categories" {
+            "detect licenses referencing non-existing categories" {
```
Comment (sschuberth) [inline model/src/test/kotlin/licenses/LicenseClassificationsTest.kt:67]: Indentation was changed.

File: model/src/test/kotlin/licenses/LicenseClassificationsTest.kt [hunk 3]
```diff
@@ -112,6 +113,14 @@ class LicenseClassificationsTest : WordSpec({
             }
         }
 
+        "contain empty categories" {
```
Comment (sschuberth) [inline model/src/test/kotlin/licenses/LicenseClassificationsTest.kt:116]: Better "maintain" instead of "contain", as the latter sounds like "any arbitrary empty categories".

File: model/src/main/kotlin/licenses/LicenseClassifications.kt [hunk 4]
```diff
@@ -56,6 +56,10 @@ data class LicenseClassifications(
...
+                putIfAbsent(category.name, mutableSetOf())
+            }
```
Comment (sschuberth) [inline model/src/main/kotlin/licenses/LicenseClassifications.kt:62]: Just double-checking that this is no oversight as you did not comment on my proposed code in any way: It's intended that you did not change this, correct? I'm really just asking, without implying that I'd prefer that you should have changed it.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: model/src/main/kotlin/licenses/LicenseClassifications.kt (hunk 4 above)
Comment (fviernau) [reply model/src/main/kotlin/licenses/LicenseClassifications.kt:62]: Yes, I intentionally haven't changed that.

File: model/src/main/kotlin/licenses/LicenseClassifications.kt (hunk 1 above)
Comment (fviernau) [reply model/src/main/kotlin/licenses/LicenseClassifications.kt:60]: ok.

**Review by sschuberth — APPROVED** [review]
